### PR TITLE
[21] Add Contributing Guide

### LINKED ISSUE #21: Add Contributing Guide
Probably can mostly copy the one from Reservations!

Should also write up what all the labels are for, milestone/project convention.

### PR DESCRIPTION
**Pull request by AlanLiu96**
Resolves #21
	- Adds a contributing guide (mainly from reservations)
	- Todo: add gh-pages for documentation

### CODE REVIEW

**Comment by AlanLiu96** [comment]
Mainly just copied over the Reservations one. I left in the old email for stc-reservations. Other than that, I just switched over all Reservations to Vesta and updated tags. It does mention a gh-pages branch, which isn't set up yet, but I'm making an issue for it right now

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
+# Contributing to Vesta
+Thanks so much for helping with the development of Vesta! This guide
+will cover the process for helping out with the project and should allow you
+to get started quickly and easily. If you need any help with contributing,
+please e-mail `stc-reservations{at}yale{dot}edu` or mention
```
Comment (esoterik) [inline CONTRIBUTING.md]: we do actually have a (general) rails team email, so we should probably use that. It's `stc-developers-rails@elilists.yale.edu`. (Not everyone is on it right now, but I'll fix that)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+including but not limited to the use case, a proposed implementation, and why
+it is likely to be useful.
+
+## Improving Documentation
+One great way of helping out is improving documentation, which can take the
+form of user-, developer- , and sysadmin-facing documentation. Our user-facing
+documentation is located on our GitHub Pages [site](https://yalestc.github.io/vesta),
+while our developer- and sysadmin-facing documentation is located in code
+comments, in our [README](https://github.com/YaleSTC/vesta/blob/master/README.md),
+in the `/doc` folder of the repo, and in our [GitHub wiki](https://github.com/YaleSTC/vesta/wiki).
```
Comment (esoterik) [inline CONTRIBUTING.md]: should we maybe take this out and add it back when we actually have this documentation?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+The easiest way to make small improvements to our user-facing documentation is
+by editing the file with the online GitHub editor: browse the
+[`gh-pages` branch](https://github.com/YaleSTC/vesta/tree/gh-pages),
+choose the file you want to edit, click the "Edit" button, make changes, and
+follow the GitHub instructions from there. Once your changes are merged they
+will immediately appear on the website.
+
+If you'd like to help comment our codebase to make it more easily
+understandable, we ideally try to use [YARD](https://github.com/lsegal/yard)
```
Comment (esoterik) [inline CONTRIBUTING.md]: should change "ideally try to use" to just "use!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+  * If you don't have an issue to reference, make sure you don't need to open
+    one before opening the PR.
+* Use bullet points on the following lines to elaborate on your brief
+  description.
+
+Example:
+```
+Fix the very important thing that was broken
+
+closes #1234
```
Comment (esoterik) [inline CONTRIBUTING.md]: Change "closes" to "Resolves"

**Comment by esoterik** [comment]
rebase and this is ready to merge!

**Pull request merged**